### PR TITLE
Stripe offer signals

### PR DESCRIPTION
**Pull request by fahzee1**
[TC-3652](https://whitemoondreams.atlassian.net/browse/TC-3652) - Offer Signals

### CODE REVIEW

File: vendor/signals.py [hunk 1]
```diff
@@ -63,4 +65,120 @@ def stripe_delete_customer_signal(sender, instance, **kwargs):
...
+        return None
+
+    if instance.has_any_discount_or_trial():
+        coupon_data = processor.build_coupon(instance, DEFAULT_CURRENCY)
+
+        stripe_coupon = processor.stripe_create_object(processor.stripe.Coupon, coupon_data)
+        instance.meta['stripe']['coupon_id'] = stripe_coupon.id
+
+    instance.save()
+    logger.info(f"stripe_create_offer_signal instance: offer {instance.pk} successfully created on Stripe")
```
Comment (rhimmelbauer) [inline vendor/signals.py:160]: cant you just replace this with the function we already for for creating offers?


`processor.create_offers([offer])`

File: vendor/signals.py [hunk 2]
```diff
@@ -63,4 +65,120 @@ def stripe_delete_customer_signal(sender, instance, **kwargs):
...
+
+        instance.save()
+
+        if len(coupon_matches) > 1:
+            logger.warning(
+                f"stripe_create_offer_signal: more than one coupon found for offer: {instance.pk} on site: {instance.site.domain}")
+
+        if len(query_result.data) > 1:
+            logger.warning(
+                f"stripe_create_offer_signal: more than one offer/product found for pk: {instance.pk} on site: {instance.site.domain}")
```
Comment (rhimmelbauer) [inline vendor/signals.py:129]: cant you just replace this with the function we already for for creating offers?


`processor.update_offers([offer])`

**Review by rhimmelbauer — COMMENTED** [review]
Actually I think it would be wise to do the same thing with the customer signal.

File: vendor/signals.py (hunk 1 above)
Comment (fahzee1) [reply vendor/signals.py:160]: ahh

**Pull request merged**